### PR TITLE
hotfix to throw http 404 error for api calls other than init for new users

### PR DESCRIPTION
**Pull request by ppratikcr7**
This resolves an issue caught during QA of ticket #2045 where we got 404 httperror for hitting init api call for new users. This will filter the init api call to not throw 404 error on user not found.

### CODE REVIEW

File: backend/packages/Upgrade/src/api/services/ExperimentUserService.ts [hunk 1]
```diff
@@ -298,14 +298,22 @@ export class ExperimentUserService {
     return this.userRepository.save(newDocument);
   }
 
-  public async getUserDoc(experimentUserId, logger): Promise<RequestedExperimentUser> {
+  public async getUserDoc(
+    experimentUserId: string,
+    logger: UpgradeLogger,
+    api?: string
```
Comment (VivekFitkariwala) [inline backend/packages/Upgrade/src/api/services/ExperimentUserService.ts:304]: This is not a good way. The function is doing more than what the name is suggesting.

File: backend/packages/Upgrade/src/api/services/ExperimentUserService.ts [hunk 2]
```diff
@@ -298,14 +298,22 @@ export class ExperimentUserService {
...
+  ): Promise<RequestedExperimentUser> {
     const experimentUserDoc = await this.getOriginalUserDoc(experimentUserId, logger);
     if (experimentUserDoc) {
       const userDoc = { ...experimentUserDoc, requestedUserId: experimentUserId };
       logger.info({ message: 'Got the user doc', details: userDoc });
       return userDoc;
     } else {
-      throw new HttpError(404, `Experiment User not found: ${experimentUserId}`);
```
Comment (VivekFitkariwala) [inline backend/packages/Upgrade/src/api/services/ExperimentUserService.ts:308]: The error should not be thrown from here

File: backend/packages/Upgrade/src/api/controllers/ExperimentClientController.v5.ts [hunk 3]
```diff
@@ -179,7 +179,7 @@ export class ExperimentClientController {
   ): Promise<Pick<ExperimentUser, 'id' | 'group' | 'workingGroup'>> {
     request.logger.info({ message: 'Starting the init call for user' });
     // getOriginalUserDoc call for alias
-    const experimentUserDoc = await this.experimentUserService.getUserDoc(experimentUser.id, request.logger);
+    const experimentUserDoc = await this.experimentUserService.getUserDoc(experimentUser.id, request.logger, 'init');
```
Comment (VivekFitkariwala) [inline backend/packages/Upgrade/src/api/controllers/ExperimentClientController.v5.ts:182]: Create a function in the controller which should return the experiment user doc if exist else should throw an error

**Review by VivekFitkariwala — CHANGES_REQUESTED** [review]

**Comment by danoswaltCL** [comment]
if this is a bugfix for release/6.0, it should be based from that

**Comment by ppratikcr7** [comment]
> if this is a bugfix for release/6.0, it should be based from that

No, this was an enhancement kind of ticket, which was intended for proper errors codes from api's. This was intended to go into dev.

**Review by VivekFitkariwala — APPROVED** [review]

**Pull request merged**